tests/e2e/hcp_machine_pool_test.go

```diff
@@ -406,7 +410,166 @@ var _ = Describe("HCP Machine Pool", labels.Feature.Machinepool, func() {
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("Expected key=value:scheduleType format for taints. Got 'v'"))
+
+			By("with taint no schedule type")
+			_, err = machinePoolService.CreateMachinePool(clusterID, "anything", "--replicas", "2", "--taints", "k=v")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("Expected key=value:scheduleType format for taints. Got 'k=v'"))
+
+			By("with taint empty schedule type")
+			_, err = machinePoolService.CreateMachinePool(clusterID, "anything", "--replicas", "2", "--taints", "k=v:")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("Expected a not empty effect"))
+
+			By("with auto-repair wrong value")
+			_, err = machinePoolService.CreateMachinePool(clusterID, "anything", "--replicas", "2", "--autorepair=v")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("strconv.ParseBool: parsing \"v\": invalid syntax"))
+
+			By("with not supported version")
+			_, err = machinePoolService.CreateMachinePool(clusterID, "anything", "--replicas", "2", "--version", "4.12.1")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("Expected a valid OpenShift version"))
+
+			By("with not supported version")
+			_, err = machinePoolService.CreateMachinePool(clusterID, "anything", "--replicas", "2", "--multi-availability-zone")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).
+				Should(
+					ContainSubstring("Setting `multi-availability-zone` flag is not supported for HCP clusters"))
+		})
+
+		It("deletion - [id:56783]", labels.Medium, labels.Runtime.Day2, func() {
+			By("with no machinepool id")
+			_, err := machinePoolService.DeleteMachinePool(clusterID, "")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("You need to specify a machine pool name"))
+
+			By("with non existing machinepool id")
+			_, err = machinePoolService.DeleteMachinePool(clusterID, "anything")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("Machine pool 'anything' does not exist"))
+
+			By("with invalid machinepool id")
+			_, err = machinePoolService.DeleteMachinePool(clusterID, "anything%^")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("Expected a valid identifier for the machine pool"))
+
+			By("with unknown flag --interactive")
+			_, err = machinePoolService.DeleteMachinePool(clusterID, "anything", "--interactive")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("unknown flag: --interactive"))
+
+			if !profile.ClusterConfig.MultiAZ {
+				By("Delete last remaining machinepool")
+				_, err = machinePoolService.DeleteMachinePool(clusterID, constants.DefaultHostedWorkerPool)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).Should(ContainSubstring("TODO"))
```

Comment:
Why ContainSubstring "TODO" is checked here?

Reply:
oh indeed. forgot that one...

Reply:
updated